defaults/main.yml

```diff
@@ -62,4 +63,17 @@ postgresql_dist_debian:
   conf_postgresql_src: postgresql-conf.j2
   datadir: /var/lib/postgresql/{{ postgresql_version }}/main
   basename: postgresql-{{ postgresql_version }}
+  conf_sample_file: "/usr/share/postgresql/{{ postgresql_version }}/postgresql.conf.sample"
   service: postgresql
+
+# conf_sample_file: sample configuration file, it should be copied during the installation by the installation package
+
+# the following two variables are used to download link ofr the source to get the sample configuration file
+# It will be used in case of the sample file is not find locally for any reason
```

Comment:
Any reason not to have a hard fail in case `conf_sample_file` does not exist?

Are you aware of any scenario where this file could legitimately be expected to be missing?

Reply:
On our RHEL 9 test machine, there are 4 PostgreSQL versions have been installed (13, 14, 15, 16). Only version 16 folder has the sample file and the others do not have this file.  I think the file should be there but I am not sure why it is missing.  So, I have added the logic of downloading and extracting the file to account for similar cases.

Reply:
Interesting (and slightly concerning). Note all of the PSQL sample configuration file checks seem to be currently skipped in the Molecule tests on both Rocky Linux. I don't know if that's expected or indicates another issue.

Reply:
It will check for the sample file if the original file does not exist and the configuration file has been edited before, i.e. in case of installing the same PostgreSQL version with a previous role version.